src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/actions/TransitionsAction.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.actions
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.actions.TransitionsActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.steps.AttemptTransitionStep
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.steps.Step
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+
+class TransitionsAction(
+    clusterService: ClusterService,
+    client: Client,
+    managedIndexMetaData: ManagedIndexMetaData,
+    config: TransitionsActionConfig
+) : Action(type, managedIndexMetaData) {
+
+    private val attemptTransitionStep =
+            AttemptTransitionStep(clusterService, client, config, managedIndexMetaData)
+    private val steps = listOf(attemptTransitionStep)
+
+    override fun getSteps(): List<Step> = steps
+
+    override fun getNextStep(managedIndexMetaData: ManagedIndexMetaData): Step {
+        return attemptTransitionStep
+    }
+
+    companion object {
+        const val type = "transitions"
```

Comment:
what is this action type for? and what is the difference between action type and action config type?

Reply:
Used to populate the metadata with the type of action that is being executed.
The type in this action and the type in the action config should be equal for the same actions. Will move all types out into an enum so there's no duplication.